cpp/src/aztec/honk/utils/power_polynomial.test.cpp

```diff
@@ -0,0 +1,34 @@
+#include "power_polynomial.hpp"
+#include <numeric/random/engine.hpp>
+#include <gtest/gtest.h>
+
+TEST(power_polynomial, test_full_polynomial_correctness)
+{
+    const size_t order = 17;
+    const size_t n = size_t(1) << order;
+    barretenberg::fr zeta = barretenberg::fr::random_element();
+    barretenberg::polynomial power_polynomial = honk::power_polynomial::generate_vector(zeta, n);
+
+    barretenberg::fr current_power = barretenberg::fr::one();
+    for (size_t i = 0; i < n; i++) {
+        EXPECT_EQ(power_polynomial.coefficients[i], current_power);
+        if (power_polynomial.coefficients[i] != current_power) {
+            break;
+        }
+        current_power *= zeta;
+    }
+}
+
+TEST(power_polynomial, test_evaluation_correctness)
```

Comment:
Once the Gemini PR comes in, we could test this function against the `polynomial::evaluate_mle()` in a random multi-variate point to test the correctness.